docs/index.md

```diff
@@ -1,24 +1,32 @@
 # autora-theorist-Avengers
 
-Explain what your contribution is doing here
+## Search Algorithm
 
-You can include inline mathematics by wrapping LaTeX formatted math in `$` symbols: `$(3 < 4)$` will render as:  $(3 < 4)$
+The search algorithm employed by the theorist involves fitting a quadratic model to the provided dataset using the `ParabolaRegression` class. The algorithm follows these steps:
 
-To include block mathematics wrap the expression with double `$$`:
+1. **Data Preparation**: The input data \( X \) is converted to a 2D numpy array.
+2. **Design Matrix Construction**: A design matrix \( A \) is constructed, which includes:
+   - An intercept term (a column of ones).
+   - Linear terms (the original features).
+   - Log-transformed terms (log of the original features).
+   - Quadratic terms (squares and cross-products of the features).
+3. **Coefficient Calculation**: The coefficients are calculated using the least squares method by solving the normal equation \( A \beta = y \).
+4. **Model Fitting**: The model is fitted to the data by storing the calculated coefficients.
 
-```
-$$
-E(\mathbf{v}, \mathbf{h}) = -\sum_{i,j}w_{ij}v_i h_j - \sum_i b_i v_i - \sum_j c_j h_j
-$$
-```
+The goodness of the equation is determined using statistical measures such as the coefficient of determination (R²) and mean squared error (MSE).
 
-Will render as:
+## Search Space
 
-$$
-E(\mathbf{v}, \mathbf{h}) = -\sum_{i,j}w_{ij}v_i h_j - \sum_i b_i v_i - \sum_j c_j h_j
-$$
+The search space for the theorist includes a variety of potential equations formed by different combinations of the following terms:
 
+- **Intercept Term**: A constant term.
+- **Linear Terms**: The original features of the dataset.
+- **Log-Transformed Terms**: The natural logarithm of the original features.
+- **Quadratic Terms**: The squares and cross-products of the original features.
 
-See more information on writing mathematical expressions [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/writing-mathematical-expressions). 
+The search is constrained by the structure of the design matrix, which ensures that all relevant terms are included without overfitting. The constraints include:
 
+- **Feature Selection**: All features are considered in their original, log-transformed, and quadratic forms.
+- **Equation Complexity**: The design matrix is constructed to include all necessary terms while avoiding unnecessary complexity.
 
+These constraints help in efficiently exploring the search space and identifying the best-fitting quadratic equation.
```

Comment:
I think a bit more explanation is required as to how the constraints employed on the design matrix prevent overfitting of the relevant terms.